posts/2025-10-23-docker-101/index.qmd

```diff
@@ -0,0 +1,383 @@
+---
+title: "Docker 101"
+author:
+  - "Martin Dahlö & Björn Viklund"
+date: "2025-10-23"
+date-modified: last-modified
+categories: [ "Docker" ]
+---
+
+## What is Docker?
+
+Docker containers are kind of like virtual machines; simulated/virtualised computers inside your physical computer, and you can have many of them running on the same physical computer (host computer). They have their own operating system and all, and they behave like they are normal physical computers.
+
+Unlike traditional virtual machines they don’t simulate the entire computer from the ground up, but rather create a sandboxed (i.e. isolated) environment that pretends to be a virtual machine. Since it reuses the much of the operating system already running on the computer it has almost no start-up time and very low resource usage in itself. 
+
+The main parts of docker are *images* and *containers*. The images are much like virtual disk images, the hard drives in virtual machines. It's what contains all the files. Containers are what you call it when it is running, i.e. all the processes running and memory in the "virtual machine". You can start multiple containers all running from the same image and they will be completely independent from each other.
```

Comment:
> It's what contains all the files

What is the purpose of these files? I think this sentence should be elaborated on.